src/main.rs

```diff
@@ -54,21 +54,21 @@ async fn main() -> io::Result<()> {
         .into_iter()
         .map(|b| Backend {
             address: b.url,
-            weight: b.weight,
+            weight: b.weight.unwrap_or(1),
             active_connections: Arc::new(AtomicUsize::new(0)),
             is_healthy: Arc::new(AtomicBool::new(true)),
             current_weight: Arc::new(Mutex::new(0)),
             latency: Arc::new(Mutex::new(Duration::from_secs(u64::MAX))),
         })
         .collect();
 
-    let algorithm = match config.algorithm.as_str() {
+    let algorithm = match config.algorithm.as_deref().unwrap_or("RoundRobin") {
         "RoundRobin" => RoundRobin,
         "LeastConnections" => LeastConnections,
         "WeightedRoundRobin" => WeightedRoundRobin,
         "IPHashing" => IPHashing,
         "LeastLatency" => LeastLatency,
-        _ => panic!("Unknown algorithm: {}", config.algorithm),
+        _ => panic!("Unknown algorithm: {:?}", config.algorithm),
```

Comment:
nit: maybe do it like this?
```rust
let algorithm = match config.algorithm.as_deref() {
        Some("RoundRobin") | None => RoundRobin,
        Some("LeastConnections") => LeastConnections,
        Some("WeightedRoundRobin") => WeightedRoundRobin,
        Some("IPHashing") => IPHashing,
        Some("LeastLatency") => LeastLatency,
};
```

Reply:
Sure, but maybe we should keep catch-all case, or better yet update it with an Error instead of panic?